core/templates/dev/head/components/RuleTypeSelectorDirective.js

```diff
@@ -82,7 +82,12 @@ oppia.directive('ruleTypeSelector', [function() {
         $(select2Node).select2('val', $scope.localValue);
 
         $(select2Node).on('change', function(e) {
+          $scope.localValue = e.val;
```

Comment:
I don't think you need these two lines. You just need the $scope.$apply() on line 90. Would you agree?

(I started looking into this further when I noticed that $scope.localValue was bound using '@', so I don't think this change is going to do anything.)